icu4c/source/i18n/complexunitsconverter.cpp

```diff
@@ -0,0 +1,86 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+#include <utility>
+
+#include "cmemory.h"
+#include "complexunitsconverter.h"
+#include "uassert.h"
+#include "unicode/fmtable.h"
+#include "unitconverter.h"
+
+U_NAMESPACE_BEGIN
+
+ComplexUnitsConverter::ComplexUnitsConverter(const MeasureUnit inputUnit, const MeasureUnit outputUnits,
+                                             UErrorCode &status) {
+    auto singleUnits = outputUnits.splitToSingleUnits(status);
+    MaybeStackVector<MeasureUnit> singleUnitsInOrder;
+    for (int i = 0, n = singleUnits.length(); i < n; ++i) {
+        // TODO(younies): ensure units being in order in phase 2. Now, the units in order by default.
+        singleUnitsInOrder.emplaceBack(singleUnits[i]);
+    }
+
+    ComplexUnitsConverter(inputUnit, std::move(singleUnitsInOrder), status);
+}
+
+ComplexUnitsConverter::ComplexUnitsConverter(const MeasureUnit inputUnit,
+                                             const MaybeStackVector<MeasureUnit> outputUnits,
+                                             UErrorCode &status) {
+    if (outputUnits.length() == 0) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
+        return;
+    }
+
+    MaybeStackVector<UnitConverter> converters;
+    for (int i = 0, n = outputUnits.length(); i < n; i++) {
+        if (i == 0) { // first element
+            converters.emplaceBack(UnitConverter(inputUnit, *outputUnits[i], status));
```

Comment:
You don't need `UnitConverter` in this line (also below).

Reply:
Done!